Campus Density/SectionControllers/MenuSectionController.swift

```diff
@@ -31,28 +31,28 @@ class MenuSectionController: ListSectionController {
     // Note: not currently in use as there should not be a gap between menu and hours in covid era flux
     func findLongestMenu(menuModel: MenuModel) -> CGFloat {
         var maxHeight: CGFloat = 0
-        for meal in menuModel.mealNames {
-            let menuHeight = heightForMeal(meal: meal)
+        for menu in menuModel.menu.menus {
+            let menuHeight = heightForMenu(menuData: menu)
             maxHeight = CGFloat.maximum(maxHeight, menuHeight)
         }
         return maxHeight
     }
 
     /// The height of the menu cell for a specific `Meal`
-    func heightForMeal(meal: Meal) -> CGFloat {
+    func heightForMenu(menuData: MenuData) -> CGFloat {
         guard let containerSize = collectionContext?.containerSize else { return .zero }
-        let width = containerSize.width - 4 * Constants.smallPadding // 2 each for the exterior cell and interior cell
-        let menuHeight = ceil(MenuInteriorCell.getMenuString(todaysMenu: menuModel.menu, selectedMeal: meal).boundingRect(with: CGSize.init(width: width, height: 0), options: [.usesLineFragmentOrigin, .usesFontLeading], context: nil).size.height) + 2 * Constants.smallPadding
+        let width = containerSize.width - 2 * Constants.smallPadding
+        let menuHeight = ceil(MenuInteriorCell.getMenuString(menuData: menuData).boundingRect(with: CGSize.init(width: width, height: 0), options: [.usesLineFragmentOrigin, .usesFontLeading], context: nil).size.height)
         return menuHeight
     }
 
     override func sizeForItem(at index: Int) -> CGSize {
         guard let containerSize = collectionContext?.containerSize else { return .zero }
-        let meal = menuModel.selectedMeal // Swiping from a small meal to a larger one gives the warning
+        let meal = menuModel.selectedMeal
         if meal == .none {
             return CGSize(width: containerSize.width, height: unavailableText.height(withConstrainedWidth: containerSize.width, font: .eighteenBold))
         } else {
-            return CGSize(width: containerSize.width, height: heightForMeal(meal: meal)) // normally use tallestMenu for height
+            return CGSize(width: containerSize.width, height: tallestMenu + (MenuInteriorCell.hoursLabelHeight + Constants.smallPadding)) // use heightForMenu() to change size every time
```

Comment:
I'm just a bit concerned about whether this should be the 'right' way to pass on information. What you did makes complete sense, but I'm just wary about making sizes static from the child components and then directly accessing those dimensions in the parent. I doubt it will matter too much in our case, but static variables often do cause memory leaks (and hence you'll see them often used in singleton patterns, but not extensively anywhere else). Looking at other parts of how SectionControllers are laid out, I'd say the pattern I see is to define the cell height in the section controller itself, and then laying out child components in the corresponding `*Cell.swift` file to fit accordingly

Reply:
Hm sure, I can make it so the constant is stored in the section controller. I think I was trying to avoid the previous pattern, which was a lot of magic numbers/duplicated constants everywhere, aka storing hoursLabelHeight here in both the section controller and cell (so much for being DRY). However if we just relied on fitting the cell to the size of the section controller I'm not sure how we'd give the hours label/clock symbol a certain height if desired without repeating ourselves. Since there are menus and hours label in the cell, and the superview height is just one number. I did not think about memory leaks as I assumed static let was safe enough to be replaced during during compile-time like in Java.

Reply:
Yeah. I completely agree about unwanted magic numbers in our code. Either way, I think this code is mergeable, but we should try to find workarounds

Reply:
Might be a consequence of this entire app being based around IGListKit when it really shouldn't be and us having to make SectionControllers for everything

Reply:
You're right. Either way, it might not be worth our time to rework this right now, that will require some significant re-architecting. We should keep this in mind for Flux Fitness. Let's get this merged!